src/compiler/watch.ts

```diff
@@ -75,7 +75,7 @@ namespace ts {
             } :
             (diagnostic, newLine, options) => {
                 clearScreenIfNotWatchingForFileChanges(system, diagnostic, options);

```

Comment:
may be let ```clearScreenIfNotWatchingForFiles``` send you the return value if the screen is cleared since it is not cleared in ```options.extendedDiagostics || options.diagnostics``` too.